ibis/backends/exasol/__init__.py

```diff
@@ -278,12 +278,14 @@ def process_item(item: Any):
         with self._safe_raw_sql(create_stmt_sql):
             if not df.empty:
                 self.con.ext.insert_multi(name, rows)
-        atexit.register(self._clean_up_tmp_table, ident)
 
-    def _clean_up_tmp_table(self, ident: sge.Identifier) -> None:
-        with self._safe_raw_sql(
-            sge.Drop(kind="TABLE", this=ident, exists=True, cascade=True)
-        ):
+    def _register_memtable_finalizer(self, op: ops.InMemoryTable):
+        weakref.finalize(op, self._clean_up_tmp_table, op.name)
```

Comment:
Side nit: `exasol` sets `supports_in_memory_tables = True` at the top, but since it redefines all the memtable methods should probably just delete that flag.